src/assets/js/canvasjs.react.js

```diff
@@ -1,48 +1,48 @@
-		this.containerProps.height = props.containerProps && props.containerProps.height ? props.containerProps.height : this.options.height ? this.options.height + "px" : "400px";
-		this.chartContainerId = "canvasjs-react-chart-container-" + CanvasJSChart._cjsContainerId++;
-	}	
-	componentDidMount() {
-		//Create Chart and Render		
-		this.chart = new CanvasJS.Chart(this.chartContainerId, this.options);
-		this.chart.render();
-		
-		if(this.props.onRef)
-			this.props.onRef(this.chart);
-	}	
-    shouldComponentUpdate(nextProps, nextState){
-		//Check if Chart-options has changed and determine if component has to be updated
-        return !(nextProps.options === this.options);
-    }
-	componentDidUpdate() {
-		//Update Chart Options & Render
-		this.chart.options = this.props.options;
-		this.chart.render();
-	}
-	componentWillUnmount() {
-		//Destroy chart and remove reference
-		this.chart.destroy();
-		if(this.props.onRef)
-			this.props.onRef(undefined);
-	}		
-	render() {		
-		//return React.createElement('div', { id: this.chartContainerId, style: this.containerProps });		
-		return <div id = {this.chartContainerId} style = {this.containerProps}/>		
-	}	
-}
-
-var CanvasJSReact = {
-    CanvasJSChart: CanvasJSChart,
-    CanvasJS: CanvasJS
-};
-
+var React = require('react');
+var CanvasJS = require('./canvasjs.min');
+CanvasJS = CanvasJS.Chart ? CanvasJS : window.CanvasJS;
+
+class CanvasJSChart extends React.Component {
+	static _cjsContainerId = 0
+	constructor(props) {		
+		super(props);		
+		this.options = props.options ? props.options : {};		
+		this.containerProps = props.containerProps ? props.containerProps : {width: "100%", position: "relative"};
+		this.containerProps.height = props.containerProps && props.containerProps.height ? props.containerProps.height : this.options.height ? this.options.height + "px" : "400px";
+		this.chartContainerId = "canvasjs-react-chart-container-" + CanvasJSChart._cjsContainerId++;
+	}	
+	componentDidMount() {
+		//Create Chart and Render		
+		this.chart = new CanvasJS.Chart(this.chartContainerId, this.options);
+		this.chart.render();
+		
+		if(this.props.onRef)
+			this.props.onRef(this.chart);
+	}	
+    shouldComponentUpdate(nextProps, nextState){
+		//Check if Chart-options has changed and determine if component has to be updated
+        return !(nextProps.options === this.options);
+    }
+	componentDidUpdate() {
+		//Update Chart Options & Render
+		this.chart.options = this.props.options;
+		this.chart.render();
+	}
+	componentWillUnmount() {
+		//Destroy chart and remove reference
+		this.chart.destroy();
+		if(this.props.onRef)
+			this.props.onRef(undefined);
+	}		
+	render() {		
+		//return React.createElement('div', { id: this.chartContainerId, style: this.containerProps });		
+		return <div id = {this.chartContainerId} style = {this.containerProps}/>		
+	}	
+}
+
+var CanvasJSReact = {
+    CanvasJSChart: CanvasJSChart,
+    CanvasJS: CanvasJS
+};
+
```

Comment:
what happened with this file?

Reply:
Great question, I'm looking at the difference between this branch and what's currently on the master branch and don't notice a difference. Not sure why it's displayed here.

Reply:
Probably worth rebasing to get changes to this file out of the history, or as we are squashing you might just be able to have it cancel out. Removing "bad commit" via interactive rebase might also help; it could just be like line ending weridness or something and it's probably best to make sure this remains untouched.